src/BaseFirestoreRepository.ts

```diff
@@ -50,38 +50,56 @@ export class BaseFirestoreRepository<T extends IEntity>
   }
 
   async create(item: T): Promise<T> {
-    if (item.id) {
-      const found = await this.findById(item.id);
-      if (found) {
-        return Promise.reject(
-          new Error(`A document with id ${item.id} already exists.`)
-        );
+    try {
+      if (item.id) {
+        const found = await this.findById(item.id);
+        if (found) {
+          throw `A document with id ${item.id} already exists.`;
+        }
+      };
+       
+      const errors = await this.validate(item);
+
+      if (errors.length) {
+        throw errors;
       }
-    }
 
-    const doc = item.id
-      ? this.firestoreColRef.doc(item.id)
-      : this.firestoreColRef.doc();
+      const doc = item.id
+        ? this.firestoreColRef.doc(item.id)
+        : this.firestoreColRef.doc();
 
-    if (!item.id) {
-      item.id = doc.id;
-    }
+      if (!item.id) {
+        item.id = doc.id;
+      }
+  
+      await doc.set(this.toSerializableObject(item));
 
-    await doc.set(this.toSerializableObject(item));
+      if (this.collectionType === FirestoreCollectionType.collection) {
+        this.initializeSubCollections(item);
+      }
 
-    if (this.collectionType === FirestoreCollectionType.collection) {
-      this.initializeSubCollections(item);
+      return item;
+    } catch (error) {
+      throw new Error(error);
```

Comment:
Isn't the error in the catch already an instance of `Error`? Why wrapping it in another `Error`?

Reply:
I'll take a look at this when adding the new error constructors :)